ddl/ddl_api.go

```diff
@@ -4354,6 +4357,10 @@ func (d *ddl) getModifiableColumnJob(ctx context.Context, sctx sessionctx.Contex
 		return nil, errors.Trace(err)
 	}
 
+	if err = checkTooBigFieldLengthAndTryAutoConvert(&newCol.FieldType, newCol.Name.O, sctx.GetSessionVars()); err != nil {
```

Comment:
Could we put it to `setCharsetCollationFlenDecimal`? Then we needn't call it in two places, and we can do `adjustBlobTypesFlen` in one function.

Reply:
Resolved, thanks.